### PR TITLE
Update Colossal AI docs and integration

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

The colossal-ai repo is now public: https://github.com/Lightning-AI/lightning-colossalai and pending a pypi release.
The readme + documentation gets updated on both sides and getting cross-linked.

cc @borda @carmocca @akihironitta

### CODE REVIEW

File: docs/source-pytorch/advanced/model_parallel/model_parallel.rst [hunk 1]
```diff
@@ -52,133 +52,18 @@ Sharding techniques help when model sizes are fairly large; roughly 500M+ parame
...
-    model = MyModel()
-    my_strategy = ColossalAIStrategy(placement_policy="cpu")
-    trainer = Trainer(accelerator="gpu", devices=4, precision=16, strategy=my_strategy)
-    trainer.fit(model)
-
-.. _sharded-training:
-
-****************
-Sharded Training
-****************
```
Comment (awaelchli) [inline docs/source-pytorch/advanced/model_parallel/model_parallel.rst:157]: This removal was missed

File: docs/source-pytorch/advanced/model_parallel/colossalai.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+.. code-block:: python
+
+    from lightning_colossalai import ColossalAIStrategy
+
+    strategy = ColossalAIStrategy(...)
+    trainer = Trainer(strategy=strategy, precision=16, devices=...)
+
+
+See a full example of a benchmark with the a `GPT-2 model <https://github.com/hpcaitech/ColossalAI-Pytorch-lightning/tree/main/benchmark/gpt>`_ of up to 24 billion parameters
+
```
Comment (awaelchli) [inline docs/source-pytorch/advanced/model_parallel/colossalai.rst:47]: Reviewer: I modified this part of the text. The rest remains the same from where I copied it.

File: .azure/gpu-tests-pytorch.yml [hunk 3]
```diff
@@ -113,6 +113,8 @@ jobs:
     - bash: |
         set -e
         pip install -r requirements/pytorch/strategies.txt --find-links ${TORCH_URL}
+        # TODO: Update with pypi release once available and move to strategies.txt
+        pip install git+https://github.com/Lightning-AI/lightning-colossalai.git
```
Comment (awaelchli) [inline .azure/gpu-tests-pytorch.yml:117]: TODO before merge: waiting for package to be on pypi

**Comment by Borda** [comment]
pkg is up https://pypi.org/project/lightning-colossalai/

**Review by akihironitta — APPROVED** [review]

File: docs/source-pytorch/advanced/model_parallel/colossalai.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+.. note::
+
+    *   The only accelerator which ColossalAI supports is ``"gpu"``. But CPU resources will be used when the placement policy is set to "auto" or "cpu".
+
+    *   The only precision which ColossalAI allows is 16 (FP16).
```
Comment (carmocca) [inline docs/source-pytorch/advanced/model_parallel/colossalai.rst:52]: Mixed. What happens to the trainer flag if it's set to something else? Is it ignored, does it error out?

**Review by carmocca — APPROVED** [review]

File: docs/source-pytorch/advanced/model_parallel/colossalai.rst (hunk 4 above)
Comment (awaelchli) [reply docs/source-pytorch/advanced/model_parallel/colossalai.rst:52]: Yes, the validation happens in the precision plugin and it errors out saying the only supported one is 16.

Comment (awaelchli) [reply docs/source-pytorch/advanced/model_parallel/colossalai.rst]: clarified that it is mixed precision

**Review by Borda — APPROVED** [review]

**Pull request merged**